src/components/Molecules/TransactionsList/TransactionsList.tsx

```diff
@@ -0,0 +1,356 @@
+import { type Option, None, Some } from "@/utils/option";
+import { type ChainItem, calculatePrettyBalance } from "@covalenthq/client-sdk";
+import { useEffect, useState } from "react";
+import {
+    type ColumnDef,
+    type SortingState,
+    flexRender,
+    getCoreRowModel,
+    getSortedRowModel,
+    useReactTable,
+} from "@tanstack/react-table";
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableHeader,
+    TableRow,
+} from "@/components/ui/table";
+import { Checkbox } from "@/components/ui/checkbox";
+import { TokenAvatar } from "../../Atoms/TokenAvatar/TokenAvatar";
+import { timestampParser } from "@/utils/functions";
+import { TableHeaderSorting } from "@/components/ui/tableHeaderSorting";
+import { GRK_SIZES } from "@/utils/constants/shared.constants";
+import { useCovalent } from "@/utils/store/Covalent";
+import { SkeletonTable } from "@/components/ui/skeletonTable";
+import {
+    type CrossChainTransaction,
+    type TransactionListProps,
+} from "@/utils/types/molecules.types";
+import { Address } from "@/components/Atoms/Address/Address";
+
+export const TransactionsList: React.FC<TransactionListProps> = ({
+    chain_names,
+    address,
+}) => {
+    const { covalentClient, chains } = useCovalent();
+
+    const [sorting, setSorting] = useState<SortingState>([
+        {
+            id: "block_signed_at",
+            desc: true,
+        },
+    ]);
+    const [rowSelection, setRowSelection] = useState({});
+    const [maybeResult, setResult] =
+        useState<Option<CrossChainTransaction[]>>(None);
+    const [error, setError] = useState({ error: false, error_message: "" });
+    const [filterResult, setFilterResult] =
+        useState<Option<CrossChainTransaction[]>>(None);
+
+    useEffect(() => {
+        (async () => {
+            setResult(None);
+            const promises = chain_names.map(async (chain_name) => {
```

Comment:
Lets change this to single chain for now

Reply:
d71da43